common/interactive-object/server.mjs

```diff
@@ -239,6 +240,25 @@ class InteractiveObjectServerBaseClass {
     return this.sf_exit(playerID, {next: this.dataStore.get(playerID)});
   }
 
+  async sf_textBox(playerID, kwargs) {
```

Comment:
`textBox` maybe confusing for what this method does. Consider naming it "showDialogAndCheckKey" or similar?